src/Data/Term.hs

```diff
@@ -61,14 +67,28 @@ shift = shift' 0
 instance Show ResolveError where
   show (NotInScope x) = "Variable not in scope: " ++ x
 
+checkCollisionsHelper :: Map String Int -> Term -> Map String Int
+checkCollisionsHelper m (Var _ _) = m
+checkCollisionsHelper m (Abs x t) = let xs = findWithDefault 0 x m in checkCollisionsHelper (Map.insert x (xs + 1) m) t
+checkCollisionsHelper m (App f x) = checkCollisionsHelper (checkCollisionsHelper m f) x
+
+checkCollisions :: Term -> Set String
+checkCollisions t = fromList $ keys $ Map.filter (== 1) (checkCollisionsHelper Map.empty t)
+
+printTerm :: Set String -> Map String (IntMap Int) -> Int -> Term -> String
+printTerm s m d (Var i x) = x ++ if x `member` s then "" else "_" ++ show ((m Map.! x) IntMap.! (d - i))
```

Comment:
Following the idea of "user should be able to paste back the REPL output", it would be a good idea to modify term parser to handle underscores in identifiers